### DISCUSSION TITLE
Is there any wrong with my template `/{file}.{ext}`?

### DISCUSSION THREAD

**Discussion by tle5** [ThreeMammals/Ocelot#1938] [Q&A] [answered]
I try to set a Route in ocelot config file with information:
```json
"UpstreamPathTemplate": "/app1/{version}/{z}/{x}/{y}.{all}",
"DownstreamPathTemplate": "/{version}/{z}/{x}/{y}.png"
```
With expectation it will match with: `/app1/1/2/3/4.png`
But it doesn't work. Is there any wrong with my template?
Thank you very much.

**Comment by WeihanLi** [comment]
It seemed the `UpstreamPathTemplate` ignored the `.` character , [UrlPathPlaceholderNameAndValueFinder](https://github.com/ThreeMammals/Ocelot/blob/develop/src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs#L6)

**Reply by raman-m** [reply]
@WeihanLi  Maybe it was true for the old version in 2019...
Now I don't see `.` ignoring... Is it quite implicit thing?

I believe that `{all}` placeholder (file extension) at the end is quite strange definition too. :smile: 
There will be no issue at all if define the route like
```json
"UpstreamPathTemplate": "/app1/{version}/{z}/{x}/{file}",
"DownstreamPathTemplate": "/{version}/{z}/{x}/{file}"
```
Pay attention that the author replaces all upstream file extensions with explicit `png` one, which is wrong scenario.
I cannot get, why does the author replace all extensions with `png`? Strange!

**Comment by raman-m** [comment, marked as answer]
@tle5 
>But it doesn't work. Is there any wrong with my template?

Yes, it is wrong!
You wrote incorrect templates, and `{all}` is not presented in downstream template!
You need to define the route like this :point_down: 
```json
"UpstreamPathTemplate": "/app1/{version}/{z}/{x}/{file}",
"DownstreamPathTemplate": "/{version}/{z}/{x}/{file}"
```
Hope it helps!
